### PR TITLE
Decouple signing

### LINKED ISSUE #14: Decouple signing
https://github.com/pow-co/relay-browser-extension/blob/master/src/keys.ts#L59

this should sign with micro-btc-signer or @noble/secp256k1 directly

### PR DESCRIPTION
**Pull request by remjx**
Closes #14 

I'm not sure this is everything you asked for @relayking but at least a step in right direction

The unit test `'signs tx correctly'` gives same output with both previous and this modified version of keys.sign()

### CODE REVIEW

File: src/keys.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-import { sign, toDer } from '@relayx/crypto/lib/bitcoin/signature';
+import { toDer } from '@relayx/crypto/lib/bitcoin/signature';
 import {SIGHASH_ALL, SIGHASH_FORKID} from '@relayx/crypto/lib/bitcoin/sighash'
 import { serialize } from '@relayx/crypto/lib/bitcoin/script';
 import BufferWriter from '@relayx/crypto/lib/bitcoin/BufferWriter';
 import { KeyStorage } from "@relayx/wallet/lib/auth";
 import {get, set, clear} from './storage'
 import { getKeys } from "./crypto";
+import { sign } from '@noble/secp256k1'
+import JSBI from "jsbi";
```
Comment (relayking) [inline src/keys.ts]: Can you vendor toDer and make it work with native bigint?

**Review by relayking — CHANGES_REQUESTED** [review]

File: src/keys.ts (hunk 1 above)
Comment (remjx) [reply src/keys.ts]: vendored toDer: https://github.com/pow-co/relay-browser-extension/pull/16/commits/9bf30934b8968cec537a48c99543112b54c65243

File: src/bitcoin/utils.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    rbuf[rbuf.length - 1 - i] = buf[buf.length - 1 - i]
+  }
+  for (let i = 0; i < size - natlen; i++) {
+    rbuf[i] = 0
+  }
+  return rbuf
+}
+
+/* From @relayx/crypto src/bitcoin/utils.ts */
+export function BNToBuffer(bn: bigint, opts?: { size: number }) {
```
Comment (relayking) [inline src/bitcoin/utils.ts]: import {etc} from '@noble/secp256k'

etc.numberToBytesBE

Comment (remjx) [reply src/bitcoin/utils.ts]: updated in https://github.com/pow-co/relay-browser-extension/pull/16/commits/b2db333da31864dcc070bd81e41d2f6aeed65153

**Pull request merged**